Lib/test/_test_atexit.py

```diff
@@ -116,6 +116,17 @@ def test_bound_methods(self):
         atexit._run_exitfuncs()
         self.assertEqual(l, [5])
 
+    def test_atexit_with_unregistered_function(self):
+        # See bpo-46025 for more info
+        def func():
+            atexit.unregister(func)
+            1/0
+        atexit.register(func)
+        try:
+            atexit._run_exitfuncs()
```

Comment:
Would it make sense to use support.catch_unraisable_exception() to check if sys.unraisablehook has been called?

Reply:
👍

Reply:
Done!